File: fork-runner/src/main/java/com/shazam/fork/stat/TestExecution.java
package com.shazam.fork.stat;

import com.android.ddmlib.testrunner.TestIdentifier;
import org.jetbrains.annotations.NotNull;

public class TestExecution {

    public enum Status {
        FAILED,
        PASSED,
        IGNORED
    }

    private final TestIdentifier test;
    private final long startTime;
    private final long endTime;
    private final Status status;

    public TestExecution(TestIdentifier test, long startTime, long endTime, Status status) {
        this.test = test;
        this.startTime = startTime;
        this.endTime = endTime;
        this.status = status;
    }

    public TestIdentifier getTest() {
        return test;
    }

    public long getStartTime() {
        return startTime;
    }

    public long getEndTime() {
        return endTime;
    }

    public Status getStatus() {
        return status;
    }
}

File: fork-runner/src/main/java/com/shazam/fork/stat/TestExecution.kt
package com.shazam.fork.stat

import com.android.ddmlib.testrunner.TestIdentifier
import com.android.ddmlib.testrunner.TestResult

data class TestExecution(val test: TestIdentifier, val startTime: Long, val endTime: Long, val status: Status) {

    enum class Status {
        FAILED,
        PASSED,
        IGNORED
    }
}

fun TestResult.TestStatus.toTestExecutionStatus() = when (this) {
    TestResult.TestStatus.IGNORED -> TestExecution.Status.IGNORED
    TestResult.TestStatus.PASSED -> TestExecution.Status.PASSED
    else -> TestExecution.Status.FAILED
}